tcp_check/datadog_checks/tcp_check/tcp_check.py

```diff
@@ -107,7 +113,10 @@ def connect(self, addr, socket_type):
         with closing(socket.socket(socket_type)) as sock:
             sock.settimeout(self.timeout)
             start = get_precise_time()
-            sock.connect((addr, self.port))
+            if socket_type == socket.AF_INET6:
+                sock.connect((addr, self.port, 0, self.scope_id)) #If the address family is AF_INET6, then add the extra necessary arguments.
```

Comment:
I'm also curious why you included `scope_id` as a new config value but not also `flowinfo`?

Reply:
As also mentioned on the doc, omitting them is specific to certain methods and just for backward compatibility. In addition, they mention that it could cause problems in manipulating scoped IPv6 addresses.

Also, the method does not except None (only int).

Lastly, according to some sources, `flowinfo` does not yet have a specific functionality (it was speculated that it was added for future use) and was recommended to always set it as 0 for now.